various things that I did that I don't remember

KingGamingYT opened the pull request:
I dunno what happened, but I was able to fix almost all existing issues from my last commit.

Things fixed (for now):
 - Images will vary in size depending on which order you send them in and how many images are in the message
- Images (especially those with short heights) will have tons of dead space above and below the image, making a massive gap (screenshot included) when posted as part of a message containing other images
-  Embeds being slightly cut off vertically and/or horizontally
- Formatting breaks and images re-crop themselves if CTRL +/- is used
-  Sending 3 images will sometimes result in the bottom two images still being displayed horizontally

Things that remain:
- Videos playing within an image preview popup and video files missing their names in chat previews - This is a tough one, as it requires JS to fix, and not simple JS I've been told. The classes that used to allow these elements to display still exist, but they go unused. I just don't know how to make them work properly again. Right now, it's out of my skillset, so I'm hoping someone else will step in sometime in the future to have it fixed.

KingGamingYT committed:
various things that I did that I don't remember

I dunno what happened, but I was able to fix almost all existing issues from my last commit.

Things fixed (for now):
 - Images will vary in size depending on which order you send them in and how many images are in the message
- Images (especially those with short heights) will have tons of dead space above and below the image, making a massive gap (screenshot included) when posted as part of a message containing other images
-  Embeds being slightly cut off vertically and/or horizontally
- Formatting breaks and images re-crop themselves if CTRL +/- is used
-  Sending 3 images will sometimes result in the bottom two images still being displayed horizontally

Things that remain:
- Videos playing within an image preview popup and video files missing their names in chat previews - This is a tough one, as it requires JS to fix, and not simple JS I've been told. The classes that used to allow these elements to display still exist, but they go unused. I just don't know how to make them work properly again. Right now, it's out of my skillset, so I'm hoping someone else will step in sometime in the future to have it fixed.

KingGamingYT committed:
readme update because the plugin does work

KingGamingYT committed:
images will embed again

small fix to make it so that once the experiment rolls out to everyone, the plugin won't strip out the entirety of discord's link metadata, which would result in a 403 error when discord tries to get the image.
TL;DR: the plugin strips forced width correctly now.

KingGamingYT committed:
additional fix for video links

Video embed previews now aren't blank and don't change size when you play them.

KingGamingYT committed:
Minor changes, big fixes!

- Images (& videos) posted in groups of multiples of 3 no longer stay in a mosaic grid
- Videos posted as part of a mosaic no longer have the chance of becoming ginormous in chat
- Credited myself since I've been keeping this thing going for 4 months now!

Future plans: Making videos posted as part of a mosaic play normally in chat instead of opening up into the image viewer/carousel.

KingGamingYT committed:
Update README.md

readme update (in-progress)

KingGamingYT committed:
Update README.md

again

KingGamingYT committed:
big readme update!! (actually shows what the plugin does wow)

KingGamingYT committed:
video fix!

-Videos no longer have a letterboxed border around them and should display in chat properly.

NEXT UP: Making videos posted as part of a mosaic play inline in chat _instead_ of in the image carousel.

KingGamingYT committed:
readme typo oopsie

KingGamingYT committed:
small fix

removed a class that slightly improved video formatting in chat but majorly messed it up in the preview/carousel

KingGamingYT committed:
Fix-A-Lot update

So...sometime on the 15th of February, a ton of aspects of the plugin spontaneously broke. No idea why it happened in the first place, but most of them are now fixed!
- Images & videos that are long and/or have a high width no longer form large, clickable "gaps" below them to fit a square space.
- Videos will properly fill the screen in fullscreen mode again.

These fixes were graciously made by vantage87 on Discord, thanks a ton for their efforts in making up for my cluelessness!

TODOs:  Disabling the video carousel! Hopefully! Maybe!

KingGamingYT committed:
typo fix because I'm oblivious to my own code changes

KingGamingYT committed:
small fullscreen fix

(contributed by Vantage87 on Discord)

Tanza3D commented:
would you be okay if i archived this repo and pointed to yours? yours seems to... actually work, and prs are out of scope for me to test at the moment

KingGamingYT commented:
> would you be okay if i archived this repo and pointed to yours? yours seems to... actually work, and prs are out of scope for me to test at the moment

I wouldn't mind that at all, probably don't need to keep bugging you with pr update notifs, lol.

KingGamingYT committed:
add metadata for github and (hopefully) auto update

KingGamingYT committed:
remove updateurl in metadata

I didn't think to check beforehand and the plugin updater is a zere's plugin library module which I'm not implementing support for because I don't need it

KingGamingYT committed:
(mainly) Class Updates

Made a new branch for the version of the plugin with updated classes, as it's still unknown whether Discord plans to revert it. If the class changes become permanent, this branch will be merged with the main one.

Added a piece of code that will work in parity with the UncompressedImages plugin, given it receives patches from its dev. Until then, it's been commented out as it has no use otherwise.

KingGamingYT committed:
New branch and future addition

Made a new branch for the version of the plugin with updated classes, "class-update", as it's still unknown whether Discord plans to revert it. If the class changes become permanent, that branch will be merged with the main one.

Added a piece of code that will work in parity with the UncompressedImages plugin, given it receives patches from its dev. Until then, it's been commented out as it has no use otherwise. - Contributed by NoSkillPureAndy on Discord, known formerly as Vantage87

KingGamingYT committed:
typo fix

KingGamingYT committed:
Merge branch 'main' into class-update

KingGamingYT committed:
Merge pull request #1 from KingGamingYT/class-update

Class Updates

KingGamingYT committed:
class update

also removed a chunk duplicate code that was in there which I somehow didn't notice

KingGamingYT committed:
class update

second day in a row you love to see it

KingGamingYT committed:
re-class fix

I used the wrong class and it only applied to images instead of images and videos

KingGamingYT committed:
class fix AGAIN

they keep changing this damn class

KingGamingYT committed:
fix for single images

small addition to fix cropping of single images which are wider than they are tall.
